monetio/obs/openaq_v3.py

```diff
@@ -0,0 +1,855 @@
+
+    if all([raw is None, hourly is None, daily is None]):
+        # Default to raw
+        # FIXME: probably want hourly to be the default
+        raw = True
+        hourly = False
+        daily = False
+    else:
+        # User specified one (or more)
+        if raw is None:
+            raw = False
+        if hourly is None:
+            hourly = False
+        if daily is None:
+            daily = False
+    if sum([raw, hourly, daily]) != 1:
+        raise ValueError("exactly one of raw, hourly, daily can be True")
+    endpt_tpl = "/v3/sensors/{sensor_id}/measurements"
+    if hourly:
+        endpt_tpl += "/hourly"
+    elif daily:
+        endpt_tpl += "/daily"
+
+    # The API seems to assume local time if tz is not set.
+    # Usually we _do_ want/mean UTC, but for daily, we want to select dates,
+    # which are local time, and we don't want worry about tz differences
+    # for different sites.
+    if dates.tz is None and not daily:
+        dates = dates.tz_localize("UTC")
+    if daily and dates.tz is not None:
+        warnings.warn("ignoring tz for daily data")
+        dates = dates.tz_localize(None)
+
+    query_dt = pd.to_timedelta(query_time_split) if len(dates) > 1 else None
+    date_min, date_max = dates.min(), dates.max()
+    if query_dt is not None:
+        if query_dt <= pd.Timedelta(0):
+            raise ValueError(
+                f"query_time_split must be positive, got {query_dt} from {query_time_split!r}"
+            )
+        if date_min == date_max:
+            raise ValueError(
+                "must provide at least two unique datetimes to use query_time_split. "
+                "Set query_time_split=None to disable time splitting."
+            )
+    if hourly or raw:
+        date_max = date_max + pd.Timedelta(hours=1)
+    elif daily:
+        date_max = date_max + pd.Timedelta(days=1)
+
+    def iter_time_slices():
+        # Seems that (from <= time < to) == [from , to) is used
+        if query_dt is not None:
+            t = date_min
+            while t < date_max:
+                t_next = min(t + query_dt, date_max)
+                yield t, t_next
+                t = t_next
+        else:
+            yield date_min, date_max
+
+    # Discover locations
+    print("loading locations...")
+    meta = get_locations()
+    print(f"{len(meta)} locations detected")
+
+    # Narrow locations based on user input
+    if country is not None:
+        meta = meta.query("country_code == @country")
+    if sites is not None:
+        meta = meta.query("siteid == @sites")
+    if entity is not None:
+        raise NotImplementedError  # TODO: not sure what to use for this
+    if sensor_type is not None:
+        # FIXME: may not be the best approach
+        meta = meta.assign(
+            sensor_type=meta["is_monitor"].map(
+                {
+                    True: "reference grade",
+                    False: "low-cost sensor",
+                }
+            )
+        )
```

Comment:
@russbiggs is the isMonitor true/false breakdown the same as the old "reference grade" vs "low-cost sensor"?

Reply:
In v3 it's `monitor`. So `?monitor=true` will return just reference monitors, `?monitor=false` will return just air sensors (aka low-cost sensor). Not including this parameter will include both types.

Reply:
Thanks, seems like my approach here is correct then.

Reply:
`is_monitor` that I'm using in this code block is `isMonitor` from the locations endpoint data. I assume the the `monitor` query parameter you mention is consistent with that.